src/Microsoft.AspNetCore.Server.Kestrel/BadHttpResponseException.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+
+namespace Microsoft.AspNetCore.Server.Kestrel
+{
+    public sealed class BadHttpResponseException : InvalidOperationException
```

Comment:
I think this exception type would only be appropriate for attempting to write a body when it's not allowed by the spec. In the other cases, like modifying the headers/statuscode/callbacks too late, I think a `BadHttpResponseException` is confusing. I would prefer an `InvalidOperationException` in all cases. It's not like this is an exception type you would have want to handle specifically. It should always be avoidable.